woodwork/table_accessor.py

```diff
@@ -38,24 +38,53 @@
 dd = import_or_none('dask.dataframe')
 ks = import_or_none('databricks.koalas')
 
-
+from typing import Optional, Union
+from woodwork.logical_types import LogicalType
 class WoodworkTableAccessor:
     def __init__(self, dataframe):
         self._dataframe_weakref = weakref.ref(dataframe)
         self._schema = None
 
-    def init(self,
-             index=None,
-             time_index=None,
-             logical_types=None,
-             already_sorted=False,
-             schema=None,
-             validate=True,
-             use_standard_tags=True,
-             **kwargs):
-        """Initializes Woodwork typing information for a DataFrame.
+    def init(self, schema=None, **kwargs):
+        """Initializes Woodwork typing information for a DataFrame with only keyword arguments.
+        logical type inference is performed on columns with unspecified logical types"""
+        if schema is not None:
+            raise ValueError("call init_with_partial_schema or init_with_full_schema to pass a schema")
+        self.init_with_partial_schema(schema=None, **kwargs)
+   
+    def init_with_full_schema(self, schema: TableSchema, validate: bool = True) -> None:
+        """Initializes Woodwork typing information for a DataFrame with a complete schema"""
+        if validate:
+            _check_schema(self._dataframe, schema)
+            _check_unique_column_names(self._dataframe)
+        self._schema = schema
+
+    def init_with_partial_schema(self,
+                                 schema: Optional[TableSchema]=None,
+                                 index: Optional[str]=None,
+                                 time_index: Optional[str]=None,
+                                 logical_types: Optional[dict[str, LogicalType]]=None,
```

Comment:
Might be misreading/misunderstanding how the types work, but do we want the types for logical_types to be `Optional[dict[str, Union[str, LogicalType]]]`?

Reply:
side note: So awesome to see the types here!